services/requester/batch_tx_pool.go

```diff
@@ -106,8 +128,32 @@ func (t *BatchTxPool) Add(
 		return err
 	}
 
-	userTx := pooledEvmTx{txPayload: hexEncodedTx, nonce: tx.Nonce()}
-	t.pooledTxs[from] = append(t.pooledTxs[from], userTx)
+	// Scenarios
+	// 1. EOA activity not found:
+	// => We send the transaction individually, without adding it
+	// to the batch pool.
+	//
+	// 2. EOA activity found AND it was more than [X] seconds ago:
+	// => We send the transaction individually, without adding it
+	// to the batch pool.
+	//
+	// 3. EOA activity found AND it was less than [X] seconds ago:
+	// => We add the transaction to the batch pool, so that it gets
+	// processed and submitted according to the configured
+	// `TxBatchInterval`.
+	//
+	// For all 3 cases, we record the activity time for the next
+	// transactions that might come from the same EOA.
+	// [X] is equal to the configured `TxBatchInterval` duration.
+	lastActivityTime, found := t.eoaActivity.Get(from)
+	if !found || time.Since(lastActivityTime) > t.config.TxBatchInterval {
+		t.txChan <- hexEncodedTx
```

Comment:
Since the reader will exit when the context is canceled, this may end up blocking forever. This needs to use a select along with a signal channel that is closed when the node is shutting down.

that could be a separate `done` channel closed by `processIndividualTransactions`, or using the context if a base context is passed into the server during startup.

Reply:
Good point :100: I made some changes in https://github.com/onflow/flow-evm-gateway/pull/835/commits/bec00fea347226f7ef13f195eb941179dc349b5b . Is that close to what you suggested?